assets/shared/components/modal/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * WordPress dependencies
+ */
+import { Icon, close as closeIcon } from '@wordpress/icons';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Modal component.
+ *
+ * @param {Object}   props
+ * @param {Function} props.onClose  Callback to run when trying to close the modal.
+ * @param {string}   props.title    The title for the modal. Empty by default.
+ * @param {Object}   props.children The content of the modal.
+ * @return {JSX.Element} The modal component or null (if not open).
+ */
+const Modal = ( { onClose, title = '', children } ) => {
+	useRunOnEscape( onClose );
+
+	return (
+		<div className={ 'sensei-modal' }>
+			<button
+				className="sensei-modal sensei-modal__overlay"
+				aria-label="Close"
+				onClick={ onClose }
+			/>
+			<div className="sensei-modal__wrapper">
+				<div className="sensei-modal__header">
+					<div className="sensei-modal__title">{ title }</div>
+					<button
+						className="sensei-modal sensei-modal__close-button"
+						onClick={ onClose }
+					>
+						<Icon icon={ closeIcon } />
+					</button>
+				</div>
+				<div className="sensei-modal__content">{ children }</div>
+			</div>
+		</div>
+	);
+};
+
+const useRunOnEscape = ( onClose ) => {
+	useEffect( () => {
+		const handleEsc = ( event ) => {
+			if ( event.keyCode === 27 ) {
```

Comment:
hmmm, WDYT about maybe putting `import { ESC } from "@wordpress/keycodes"` and using that constant, `ESC`, here?

It would also be nice to verify if `event.defaultPrevented` is false, too.  Something like: `event.keyCode === ESC && !event.defaultPrevented`. 

WDYT?

Reply:
Changed to use ESCAPE here: a327b8bba

About the check for "defaultPrevented" I am not completely sure... that's supposed to prevent the default behaviour, right? But our "hide the modal" is not the default behaviour when pressing any key. So I am not sure we should check for that. Any extra opinion on this? Maybe I am missing the meaning of "preventDefault".

Reply:
About the defaultPrevented, I'm being inspired mostly by Gutenberg, which does the same thing here: https://github.com/WordPress/gutenberg/blob/1339b969bf6c700ca7ee315fbb0959bbe0767e12/packages/components/src/modal/index.js#L102-L104

I think the reasoning for using `e.defaultPrevented` vs. `e.stopPropagation()` is to still allow the event to bubble but skip its handling in the overlay if the "default behavior" is prevented. Naturally, in most situations, the default behavior here would be handled by the browser (which would do nothing, in this case), but with a child node, and in a situation where we have the event bubbling from a child node up to its parent nodes, we would recognize "preventing the default behavior" as "just skip closing the modal".

Am I making any sense? :thinking:

Reply:
I think you might be right. I finally added it when I saw it was already done in Gutenberg: 0c029ec98